llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp

```diff
@@ -5207,12 +5207,7 @@ void OpenMPIRBuilder::createIfVersion(CanonicalLoopInfo *CanonicalLoop,
   Function *F = CanonicalLoop->getFunction();
 
   // Define where if branch should be inserted
-  Instruction *SplitBefore;
-  if (Instruction::classof(IfCond)) {
-    SplitBefore = dyn_cast<Instruction>(IfCond);
-  } else {
-    SplitBefore = CanonicalLoop->getPreheader()->getTerminator();
-  }
+  Instruction *SplitBefore = CanonicalLoop->getPreheader()->getTerminator();
```

Comment:
Just out of curiosity, why this part was changed?

Reply:
The `IfCond` could happen to be defined by the allocation block where we inline the `omp.private` `alloc` regions. If we split right after `IfCond`, the private allocations will not dominate the `else` version of the loop.